### PR TITLE
feat: Test Operator with File-Based Catalog

### PR DESCRIPTION
**Pull request by adambkaplan**
Use OPM to render an OLM file-based catalog for testing. The base manifests are stored in the config/catalog directory, which contain an upgrade graph for Builds for OpenShift in its current state. This also includes a CatalogSource that can be deployed on OpenShift via Kustomize. Makefile targets and additional documentation added to help developers test the operator end to end, the way a cluster-admin will manage the operator.

### CODE REVIEW

File: config/olm/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: openshift-marketplace
+namePrefix: openshift-builds-
+
+resources:
+- catalogsource.yaml
+
+images:
```
Comment (sayan-biswas) [inline config/olm/kustomization.yaml:10]: This is not replacing the image for me.
Requires transformer config
```
images:
- kind: CatalogSource
  path: spec/image
```

**Review by avinal — CHANGES_REQUESTED** [review]
Working with Sayan's suggestion

**Comment by ramessesii2** [comment]
Thanks Adam for the detailed instructions. 🚀

File: config/olm/kustomization.yaml (hunk 1 above)
Comment (adambkaplan) [reply config/olm/kustomization.yaml:14]: Done!

**Review by avinal — APPROVED** [review]
Looks good to me. A little improvement to the docs can be removing the `$` from the commands. It makes it easier to copy commands from the Readme. We can iterate on it in a docs enhancement PR later.

**Comment by adambkaplan** [comment]
Merging based on Avinal's LGTM.

**Pull request merged**